File: Source/JavaScript/Arc/queries/for_ValidateRequestArguments/when_validating_arguments/with_null_value.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { ValidateRequestArguments } from '../../ValidateRequestArguments';

describe('when validating arguments with a null value', () => {
    let result: boolean;

    beforeEach(() => {
        result = ValidateRequestArguments('TestRequest', ['name'], { name: null });
    });

    it('should return false', () => result.should.be.false);
});

File: Source/DotNET/Chronicle.Specs/Reactors/for_CommandsResultHandler/when_checking_can_handle/with_null_value.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Arc.Chronicle.Reactors.for_CommandsResultHandler.when_checking_can_handle;

public class with_null_value : given.a_commands_result_handler
{
    bool _result;

    void Because() => _result = _handler.CanHandle(_reactorContext, null!);

    [Fact] void should_not_be_able_to_handle() => _result.ShouldBeFalse();
}
